KubeArmor/core/dockerHandler.go

```diff
@@ -223,6 +226,21 @@ func (dm *KubeArmorDaemon) UpdateDockerContainer(containerID, action string) {
 		dm.ContainersLock.Lock()
 		if _, ok := dm.Containers[containerID]; !ok {
 			dm.Containers[containerID] = container
+		} else if dm.Containers[containerID].PidNS == 0 && dm.Containers[containerID].MntNS == 0 {
```

Comment:
```
dm.ContainersLock.Lock()
if _, ok := dm.Containers[containerID]; !ok {
	dm.Containers[containerID] = container
} else if dm.Containers[containerID].PidNS == 0 && dm.Containers[containerID].MntNS == 0 {
	// this entry was updated by kubernetes before docker detects it
	// thus, we here use the info given by kubernetes instead of the info given by docker

	container.NamespaceName = dm.Containers[containerID].NamespaceName
	container.ContainerGroupName = dm.Containers[containerID].ContainerGroupName
	container.ContainerName = dm.Containers[containerID].ContainerName

	container.PolicyEnabled = dm.Containers[containerID].PolicyEnabled
	container.ProcessVisibilityEnabled = dm.Containers[containerID].ProcessVisibilityEnabled
	container.FileVisibilityEnabled = dm.Containers[containerID].FileVisibilityEnabled
	container.NetworkVisibilityEnabled = dm.Containers[containerID].NetworkVisibilityEnabled
	container.CapabilitiesVisibilityEnabled = dm.Containers[containerID].CapabilitiesVisibilityEnabled

	dm.Containers[containerID] = container

	for _, conGroup := range dm.ContainerGroups {
		if conGroup.ContainerGroupName == container.ContainerGroupName && conGroup.AppArmorProfiles[containerID] == "" {
			conGroup.AppArmorProfiles[containerID] = container.AppArmorProfile
			break
		}
	}
} else {
	dm.ContainersLock.Unlock()
	return
}
dm.ContainersLock.Unlock()
```